sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -329,6 +329,18 @@ class ExpressionParserSuite extends AnalysisTest {
       parameters = Map("error" -> "'x'", "hint" -> ": extra input 'x'"))
   }
 
+  test("function expressions with named arguments") {
+    assertEqual("encode(value => 'abc', charset => 'utf-8')",
```

Comment:
Please also add positive tests with:

* constant but non-literal values, e.g. `'utf' || '-8'` or `abs(42)`
* double parentheses in a named argument, e.g. `arg1 => ((select 1))`

Reply:
Great, thanks!